src/main/java/com/google/sps/servlets/RegistrationServlet.java

```diff
@@ -62,49 +62,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     
     String email = userService.getCurrentUser().getEmail();
     String userId = userService.getCurrentUser().getUserId();
-    
-    // Make list of selected topics, remove unchecked topics
-    List<Optional<String>> topics = new ArrayList<Optional<String>>();
-    topics.add(Optional.ofNullable(request.getParameter("math")));
-    topics.add(Optional.ofNullable(request.getParameter("physics")));
-    topics.add(Optional.ofNullable(request.getParameter("chemistry")));
-    topics.add(Optional.ofNullable(request.getParameter("biology")));
-    topics.add(Optional.ofNullable(request.getParameter("computer-science")));
-    topics.add(Optional.ofNullable(request.getParameter("social-studies")));
-    topics.add(Optional.ofNullable(request.getParameter("english")));
-    topics.add(Optional.ofNullable(request.getParameter("spanish")));
-    topics.add(Optional.ofNullable(request.getParameter("french")));
-    topics.add(Optional.ofNullable(request.getParameter("chinese")));
-    List<String> topicsToStr = topics
-                                .stream()
-                                .filter(t -> t.isPresent())
-                                .map(t -> t.get().toLowerCase())
-                                .collect(Collectors.toList());
-
-    // Add blank entry to topics list to know where default topics end and custom topics begin
-    topicsToStr.add(" ");
-
-    String otherTopics = Optional.ofNullable(request.getParameter("other")).orElse("");
-    if (!otherTopics.equals("")) {
-        // Split the list, removing commas and whitespace, and add to the rest of the topics
-        List<String> otherTopicsToList = Arrays.asList(otherTopics.split("\\s*,\\s*"));
-        for (String otherTopic : otherTopicsToList) {
-            topicsToStr.add(otherTopic);
-        }
-    }
 
     // Make entity for user with all registration info
     Entity userEntity = new Entity("User");
     createUserEntityAndPutInDatastore(datastore, userEntity, role, userId);
 
-    if(role.toLowerCase().equals("tutor")) {
+    if (role.toLowerCase().equals("both")) {
+        List<String> learningTopics = getTopics(request, "learn");
+        List<String> tutoringTopics = getTopics(request, "tutor");
```

Comment:
It took me a minute to understand why there's a "learn" list and a "tutor" list. It's so that a user can select different options for their tutor profile and student profile right? Can you write a comment clarifying this?

Reply:
Yeah, that's right. I agree that a comment would be helpful here. I'll add one rn